Automattic/course
# Refactor footer

### Introduced changes
* Refactor the footer to simplify the implementation
* Move the title to the right 
* Use another strategy to hide the title bottom


[More details::](https://github.com/Automattic/course/pull/150)
The previous approach was using height + overflow: hidden, which was breaking for sites with long titles.

I tested a lot of approaches, including the https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path but the most of stage was using the divisor block to partially cover the footer element.

This approach is not supporting to change the font-family and size because the fonts char doesn't have a precise height, so trying to solve a font family is necessary to find manually a value that is not applicable on other font families.

<img width="670" alt="Screenshot 2022-12-05 at 22 28 15" src="https://user-images.githubusercontent.com/38718/205785782-6bab6445-6f1e-4771-b841-2c8ee6991393.png">

### Extra Notes
This implementation assumes we will hide the font settings panels with PR #137

###  Testing instructions
* Go to the branch
* Check if you don't have custom settings on the footer
* Go to some page with the footer
* Check if it looks like the screenshot
* Type a block post title with multiple lines

### Screenshots
![xqB06L.png](https://user-images.githubusercontent.com/38718/205780697-33cf23d5-3b1c-4431-9869-344b5a4933c9.png)

![R6LzBs.png](https://user-images.githubusercontent.com/38718/205785999-8f5c8198-dc9a-4db8-8009-f6129e2071aa.png)
